### PR TITLE
tests: Add regression test for null set crash in best practices

### PR DESCRIPTION
**Pull request by artem-lunarg**
This PR adds regression test for the first issue (https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/4597) mentioned in: https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/4604

I'm going to investigate the second issue (https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4576) separately. That section of code was reverted at some point, but it still looks it's a good idea to add regression test because something similar might be implemented again (for example, as part of https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/4891).

### CODE REVIEW

File: tests/vklayertests_best_practices.cpp [hunk 1]
```diff
@@ -2347,3 +2347,40 @@ TEST_F(VkBestPracticesLayerTest, ImageMemoryBarrierAccessLayoutCombinations) {
         m_commandBuffer->end();
     }
 }
+
+// This is a positive test. No failures are expected.
+TEST_F(VkBestPracticesLayerTest, DrawingWithUnboundUnusedSet) {
```
Comment (spencer-lunarg) [inline tests/vklayertests_best_practices.cpp:2352]: I would just make a `tests/positive/best_practice.cpp` and move this there, I assume we will have more like this in the future

Comment (artem-lunarg) [reply tests/vklayertests_best_practices.cpp:2352]: Yes, actually it's the second positive test in this file.

**Review by spencer-lunarg — APPROVED** [review]

File: tests/positive/best_practices.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+TEST_F(VkPositiveBestPracticesLayerTest, TestDestroyFreeNullHandles) {
+    VkResult err;
+
+    TEST_DESCRIPTION("Call all applicable destroy and free routines with NULL handles, expecting no validation errors");
+
+    InitBestPracticesFramework();
+    InitState();
+
+    ASSERT_NO_FATAL_FAILURE(InitViewport());
+    ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
```
Comment (arno-lunarg) [inline tests/positive/best_practices.cpp:40]: Nit (to match your new code, which I think should be the standard)
```suggestion
    ASSERT_NO_FATAL_FAILURE(InitBestPracticesFramework());
    ASSERT_NO_FATAL_FAILURE(InitState());
    ASSERT_NO_FATAL_FAILURE(InitViewport());
    ASSERT_NO_FATAL_FAILURE(InitRenderTarget());
```

**Review by arno-lunarg — COMMENTED** [review]
LGTM!

**Review by arno-lunarg — APPROVED** [review]

File: tests/positive/best_practices.cpp (hunk 2 above)
Comment (artem-lunarg) [reply tests/positive/best_practices.cpp]: Updated.

**Pull request merged**